server/embed/etcd.go

```diff
@@ -774,3 +804,49 @@ func parseCompactionRetention(mode, retention string) (ret time.Duration, err er
 	}
 	return ret, nil
 }
+
+func (e *Etcd) setupTracing(ctx context.Context) (tracesdk.SpanExporter, *tracesdk.TracerProvider, error) {
+	exporter, err := otlp.NewExporter(ctx,
+		otlpgrpc.NewDriver(
+			otlpgrpc.WithEndpoint(e.cfg.ExperimentalDistributedTracingAddress),
+			otlpgrpc.WithInsecure(),
+		))
+	if err != nil {
+		return nil, nil, err
+	}
+	res := resource.NewWithAttributes(
+		semconv.ServiceNameKey.String(e.cfg.ExperimentalDistributedTracingServiceName),
+	)
+	// As Tracing service Instance ID must be unique, it should
+	// never use the empty default string value, so we only set it
+	// if it's a non empty string.
```

Comment:
so if Instance ID is empty string is that an error?

Reply:
As this field is optional. We just don't add it, as that is our default value, but also as an empty string can never be unique and the ID must be unique. Happy to do another approach here well!

Reply:
yeah I think driving reasonable defaults from the current member would be a nice option. If the flag is not required folks will leave it blank and then get errors later on. as noted above etcd name and or peerURL could be considered.

Reply:
this could be a followup

Reply:
etcd name is not unique per etcd instance, at least not by default (`default` :) ). But happy to do `peerURL`. 

There tend to be two approaches from what I have seen, either nothing by default or in some places, hostname is what some projects use the default.